browser/ui/brave_shields_data_controller.cc

```diff
@@ -20,6 +22,14 @@ BraveShieldsDataController::BraveShieldsDataController(
     content::WebContents* web_contents)
     : content::WebContentsObserver(web_contents) {}
 
+void BraveShieldsDataController::DidFinishNavigation(
+    content::NavigationHandle* navigation_handle) {
+  if (navigation_handle->IsInMainFrame() &&
+      !navigation_handle->IsSameDocument()) {
```

Comment:
so I guess we should also check `HasCommitted()` otherwise it will clear the counter for download links and some other edge cases. `HasCommitted()` means the url has changed or is an error page

Reply:
makes sense. added.